packages/edit-site/src/components/block-editor/index.js

```diff
@@ -211,54 +230,71 @@ export default function BlockEditor( { setIsInserterOpen } ) {
 			<SidebarInspectorFill>
 				<BlockInspector />
 			</SidebarInspectorFill>
-			<BlockTools
-				className={ classnames( 'edit-site-visual-editor', {
-					'is-focus-mode': isTemplatePart,
-				} ) }
-				__unstableContentRef={ contentRef }
-				onClick={ ( event ) => {
-					// Clear selected block when clicking on the gray background.
-					if ( event.target === event.currentTarget ) {
-						clearSelectedBlock();
-					}
-				} }
-			>
-				<BlockEditorKeyboardShortcuts.Register />
-				<BackButton />
-				<ResizableEditor
-					// Reinitialize the editor and reset the states when the template changes.
-					key={ templateId }
-					enableResizing={
-						isTemplatePart &&
-						// Disable resizing in mobile viewport.
-						! isMobileViewport
-					}
-					settings={ settings }
-					contentRef={ mergedRefs }
-				>
-					<BlockList
-						className="edit-site-block-editor__block-list wp-site-blocks"
-						__experimentalLayout={ LAYOUT }
-						renderAppender={
-							isTemplatePart && hasBlocks ? false : undefined
-						}
-					/>
-				</ResizableEditor>
-				<__unstableBlockSettingsMenuFirstItem>
-					{ ( { onClose } ) => (
-						<BlockInspectorButton onClick={ onClose } />
-					) }
-				</__unstableBlockSettingsMenuFirstItem>
-				<__unstableBlockToolbarLastItem>
-					<__unstableBlockNameContext.Consumer>
-						{ ( blockName ) =>
-							blockName === 'core/navigation' && (
-								<MaybeNavMenuSidebarToggle />
-							)
-						}
-					</__unstableBlockNameContext.Consumer>
-				</__unstableBlockToolbarLastItem>
-			</BlockTools>
+			<StyleBook.Slot>
```

Comment:
Will there be other use cases for taking over the editor canvas in the future? If so, would it be worth giving the slot a more generic name, since it's the parent of both the style book and the site editor canvas... perhaps something like `EditorCanvas.Slot`?

Reply:
> I can't think of any immediate future need for this though. Since this slot isn't exposed as a public API outside of @wordpress/edit-site I'm inclined to do the most minimal thing possible and implement a more generic API if/when the need arises.

Agreed! It's highly likely that it isn't something we'll need any time soon.

> Do you think it's worth adding an inline comment that encourages a future developer working in this area to not be afraid of a change like this? e.g. // Potentially this slot could be more generic e.g. EditorCanvas.Slot.

Yes, definitely not a blocker, but I think adding a comment like that would be good — I know I find those sorts of comments super helpful when I'm working out which things are okay for me to hack around with 😀